test/files/eddsa.iced

```diff
@@ -119,3 +119,84 @@ exports.sign_and_exchange = (T, cb) ->
+# signature (so the primary key is not "self signed").
+invalid_signed_pub = """-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+xjMEWMehfhYJKwYBBAHaRw8BAQdAkPjZ3KqkgQBYJ+QtVMyKeUtcNSnGRVCl7fqh
+zu6+/iPNB01yIFRlc3TCdgQTFgoAHgUCWMehfgIbLwMLCQcDFQoIAh4BAheAAxYC
+AQIZAQAKCRAyRxe+NnRYwkfYAQDPDO2CuFm8GWmVRBnMlXRBXEJxiPw5kyxj1OyX
+W1V5BwEAljWDLDmGj01/Kof6U8McEE4JNI0rLQ31ZdVPQG/S9AU=
+=ttjQ
+-----END PGP PUBLIC KEY BLOCK-----
+"""
+
+exports.invalid_signed_key = (T, cb) ->
+  await KeyManager.import_from_armored_pgp { armored: invalid_signed_pub }, defer err
+  T.assert err?, "importing should fail"
+  cb()
+
+#------------------------
+
+# Key is valid, but the message has invalid signature
+
+key2 = """-----BEGIN PGP PUBLIC KEY BLOCK-----
+
+xjMEWMekoBYJKwYBBAHaRw8BAQdAGXBpshF3yaBi3MPsSOR1NhRG96wLsvAEAvQT
+DUiMUBfNB01yIFRlc3TCdgQTFgoAHgUCWMekoAIbLwMLCQcDFQoIAh4BAheAAxYC
+AQIZAQAKCRBndBUOnFoQIgINAQDlFMzAKR1foEc4jDP+7ysHQEah6RNDnnIqNHAZ
+b/I/mwEAd/3i3sO5oOCTDv5TyKwDnUNtOaCNJliZPuBxuZkeAQQ=
+=AmZg
+-----END PGP PUBLIC KEY BLOCK-----
+"""
+
+msg_w_invalid_sig = """
+-----BEGIN PGP MESSAGE-----
+
+yJQCeJw7wsvMwCWWXiLKNydKQInxtGwpQ8TxJQt9U4uLE9NTFVzyU4sV/PJLFHwT
+S0pSiw7FsTCIcTGwsTKBFDFwcQrAdAqrMzJwHPzdcJ337sunM9/7fgzmfvfLevUB
+r7fWrAw8vPsrz9oyMDLM7BTUkT4y8dzUyXN3za3fH3oz9/Ssy3JiJrc58zin7bsh
+CwBfXTWS
+=JVLN
+-----END PGP MESSAGE-----
+"""
+
+exports.invalid_signature_msg = (T, cb) ->
+  await KeyManager.import_from_armored_pgp { armored: key2 }, defer err, keym
+  T.no_error err
+  await unbox { armored: msg_w_invalid_sig, keyfetch: keym }, defer err, literals, warnings
+  T.assert err?, "message should fail to verify"
+  cb()
+
+#------------------------
+
+# Minimal repro for: https://github.com/keybase/keybase-issues/issues/2787
+# Public key had multiple identities, one of which had a signature
+# that had S component of 31 bytes (instead of 32 for valid EDDSA). We
+# should be able to parse such key but fail the verification (so one
+# of the userids will be usigned)
+
+key_second_userid_invalid = """-----BEGIN PGP PUBLIC KEY BLOCK-----
+Version: Keybase OpenPGP v2.0.65
+Comment: Second userid has invalid signature (S shorter by 1 byte).
+
+xjMEWMe+KxYJKwYBBAHaRw8BAQdACovM/A9hqvpSk6ye0Dic/qhIXnACG7TvZ7Fm
+KQoLcizNGU1yIFRlc3QgPHRlc3RAa2V5YmFzZS5pbz7CdgQTFgoAHgUCWMe+KwIb
+LwMLCQcDFQoIAh4BAheAAxYCAQIZAQAKCRAk7Ki+jWkPk38dAQAA1YGs4U8erWBd
+/zuMACKX9dJbEaMpYknQpqvxYrkPvwEANOskGrr3G/kWUkIrjKnbjvwbDPifwa5V
+A73D7La5QQ7NG01yIFRlc3QgMiA8dGVzdEBrZXliYXNlLmlvPsJyBBMWCgAbBQJY
+x74rAhsvAwsJBwMVCggCHgECF4ADFgIBAAoJECTsqL6NaQ+T0TEBAMGoORS2uUF7
+kGOZKibkwoGC8irPGmBzN6tAf+ravK+HAPg7A3NA+Yqr+aTzl2xSXqIAOb2fJjIA
+AC0Xg++AATao
+=uJ/2
+-----END PGP PUBLIC KEY BLOCK-----
+"""
+
+exports.invalid_userid_sig = (T, cb) ->
+  await KeyManager.import_from_armored_pgp { armored: key_second_userid_invalid }, defer err, keym
+  T.no_error err # imported fine
+  T.assert keym.userids.length == 1 # but only one userid (of two)
```

Comment:
It should also find this other userid and verify that it indeed failed to verify, but I didn't know how